workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/AndroidDialogBounds.kt

```diff
@@ -0,0 +1,77 @@
+package com.squareup.workflow1.ui.container
+
+import android.app.Dialog
+import android.view.Gravity
+import android.view.View
+import android.view.Window
+import android.view.WindowManager
+import android.view.WindowManager.LayoutParams.FLAG_NOT_TOUCH_MODAL
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.environment
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+@WorkflowUiExperimentalApi
+public fun Dialog.setBounds(bounds: Bounds) {
+  window?.let { window ->
+    val windowRect = WindowManager.LayoutParams()
+      .apply {
+        copyFrom(window.attributes)
+        gravity = Gravity.TOP or Gravity.START
+      }
+    windowRect.width = bounds.width
+    windowRect.height = bounds.height
+    windowRect.x = bounds.left
+    windowRect.y = bounds.top
+
+    window.attributes = windowRect
+  }
+}
+
+@WorkflowUiExperimentalApi
+public fun <D : Dialog> D.maintainBounds(
+  view: View,
+  onBoundsChange: (D, Bounds) -> Unit = { d, b -> d.setBounds(b) }
+) {
+  maintainBounds(view.environment!!, onBoundsChange)
+}
+
+@WorkflowUiExperimentalApi
+public fun <D : Dialog> D.maintainBounds(
+  environment: ViewEnvironment,
+  onBoundsChange: (D, Bounds) -> Unit = { d, b -> d.setBounds(b) }
+) {
+  maintainBounds(environment[ModalArea].bounds, onBoundsChange)
+}
+
+@WorkflowUiExperimentalApi
+public fun <D : Dialog> D.maintainBounds(
+  bounds: StateFlow<Bounds>,
+  onBoundsChange: (D, Bounds) -> Unit = { d, b -> d.setBounds(b) }
+) {
+  val window = requireNotNull(window) { "Dialog must be attached to a window." }
+  window.setFlags(FLAG_NOT_TOUCH_MODAL, FLAG_NOT_TOUCH_MODAL)
```

Comment:
Wondering if this needs to be here? (I mean: one method one responsibility)

Reply:
Nice catch. At the very least I should call it out in the kdoc. I _think_ it's the right thing to do because dialogs are modal by default, and so restricting the bounds will be pointless. I'll confirm that.

WDYT?

Reply:
Moved it from here to `buildDialog`, called it out on the kdoc for `ModalScreenOverlayDialogFactory`.